src/app/components/resources/resources.component.ts

```diff
@@ -51,6 +120,38 @@ export class ResourcesComponent implements OnInit {
     modalRef.result.then(() => this.loadResources(), console.log);
   }
 
+  toggleResources = function (
+    category: string,
+    optionId: string | number,
+  ): void {
+    const updatedFilters =
```

Comment:
It would be great to make sure that all the branches in this logic have test cases.